### PR TITLE
Feature/clone

### PR DESCRIPTION
**Pull request by Orrimp**
- My first PR in Open Source world :) 
- The usage of this library in my code requires a clone method for frame and its items. 
- Wrote a test for cloning of frame
- Stupid implementaiton of Clone for Item. **Take any suggestions!**
- Refactored lib.rs into errors.rs and user.rs to increase readability. 
- We need a rustfmt to ensure expected formatting before PR.

### CODE REVIEW

**Comment by mr-sven** [comment]
I'm checking your code, in basic it looks good.  For my point of view rustfmt is missing a lot of features, so I use it only manually.

File: src/item.rs [hunk 1]
```diff
@@ -437,7 +461,7 @@ pub fn write_timestamp<W: Write>(writer: &mut W, date_time: &DateTime<Utc>) -> R
 pub fn read_timestamp<R: Read>(reader: &mut R) -> Result<DateTime<Utc>> {
     let seconds = reader.read_le::<i64>()?;
     let nanos = reader.read_le::<u32>()?;
-    Ok(DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(seconds, nanos), Utc))
```
Comment (mr-sven) [inline src/item.rs:440]: NaiveDateTime::from_timestamp is deprecated and from_timestamp_opt should be used.

**Review by mr-sven — APPROVED** [review]
Updating the PR, for me it's ok now.

File: src/item.rs (hunk 1 above)
Comment (Orrimp) [reply src/item.rs:440]: Hi, I a bit confused. I am not using from_timestamp_opt? 
Was this a self note/

**Comment by Orrimp** [comment]
Could we merge this with the main branch or is there still work to be done?

**Pull request merged**